Adding Maestro as a suggestion of E2E test section (#3719)
facebook/react-native-website: docs/testing-overview.md

Try to cover the vital parts of your app with E2E tests: authentication flow, core functionalities, payments, etc. Use faster JS tests for the non-vital parts of your app. The more tests you add, the higher your confidence, but also, the more time you'll spend maintaining and running them. Consider the tradeoffs and decide what's best for you.

> There are several E2E testing tools available: in the React Native community, [Detox](https://github.com/wix/detox/) is a popular framework because it’s tailored for React Native apps. Another popular library in the space of iOS and Android apps is [Appium](http://appium.io/), but if you're looking for a simple way to create E2E tests, you can look [Maestro](https://maestro.mobile.dev/) library.

Reviewer: ```suggestion
There are several E2E testing tools available: in the React Native community, [Detox](https://github.com/wix/detox/) is a popular framework because it’s tailored for React Native apps. Another popular library in the space of iOS and Android apps is [Appium](http://appium.io/) or [Maestro](https://maestro.mobile.dev/).
```

> There are several E2E testing tools available: in the React Native community, [Detox](https://github.com/wix/detox/) is a popular framework because it’s tailored for React Native apps. Another popular library in the space of iOS and Android apps is [Appium](http://appium.io/), but if you're looking for a simple way to create E2E tests, you can look [Maestro](https://maestro.mobile.dev/) library.

Reviewer 2: ```suggestion
There are several E2E testing tools available: in the React Native community, [Detox](https://github.com/wix/detox/) is a popular framework because it’s tailored for React Native apps. Another popular library in the space of Android and iOS apps is [Appium](http://appium.io/), but if you're looking for a simple way to create E2E tests, you can look [Maestro](https://maestro.mobile.dev/) library.
```

Not strictly related to this PR, but let's also reorder the platforms. 🙂

Reviewer: Please apply this suggestion @NandoMB and we can merge this 👍

Reviewer 2: I can follow up with platform reorder in separate PR, let's merge this one as-is, so it won't become stale.
